src/Node/Internal.hs

```diff
@@ -366,7 +386,7 @@ initialStatistics = do
 data HandlerProvenance peerData m t =
       -- | Initiated locally, _to_ this peer. The Nonce is present if and only
       --   if it's a bidirectional connection.
-      Local !NT.EndPointAddress (Maybe (Nonce, SharedExclusiveT m peerData, NT.ConnectionBundle, t))
+      Local !NT.EndPointAddress (Maybe (Nonce, SharedExclusiveT m peerData, NT.ConnectionBundle, Promise m (), t))
```

Comment:
I would really suggest using datatype here, guessing which meaning every element of tuple corresponds to is really cumbersome.